terraform/postgresql/provision/data.tf

```diff
@@ -39,9 +39,22 @@ locals {
 
   subnet_group = length(var.rds_subnet_group) > 0 ? var.rds_subnet_group : aws_db_subnet_group.rds-private-subnet[0].name
 
-  max_allocated_storage = (var.storage_autoscale && var.storage_autoscale_limit_gb > var.storage_gb) ? var.storage_autoscale_limit_gb : null
+  should_limit_autoscale_storage = var.storage_autoscale && var.storage_autoscale_limit_gb > var.storage_gb
+  max_allocated_storage          = local.should_limit_autoscale_storage ? var.storage_autoscale_limit_gb : null
 
-  rds_vpc_security_group_ids = length(var.rds_vpc_security_group_ids) == 0 ? [aws_security_group.rds-sg[0].id] : split(",", var.rds_vpc_security_group_ids)
+  rds_vpc_security_group_ids = length(var.rds_vpc_security_group_ids) == 0 ? [
+    aws_security_group.rds-sg[0].id
+  ] : split(",", var.rds_vpc_security_group_ids)
+
+  is_maintenance_window_blank = length(compact([
```

Comment:
I was wondering whether you'd considered writing this in [HIL](https://github.com/hashicorp/hil). The broker [interpolates HIL](https://github.com/cloudfoundry/cloud-service-broker/blob/280aaaa3664d0bbd36c54fb52a48c05d7f5b4fc3/docs/brokerpak-specification.md?plain=1#L440) and while it's an awkward language, we could write tests for it more easily than we can for expressions written in Terraform HCL.

Reply:
Apparently, I picked up some old documentation [claiming that conditionals aren't supported](https://github.com/hashicorp/hil). I've found the more recent docs, and will look into that.

Reply:
It may be a bit too clumsy to fit in a single attribute though.

Reply:
Unfortunately, as I have double-checked, the way we deal with null values removes them completely from the interpolation context, resulting in the following error when trying to create an instance:

> couldn't compute the value for "maintenance_window", template: "${maintenance_day}:${maintenance_start_hour}:${maintenance_start_min}-${maintenance_day}:${maintenance_end_hour}:${maintenance_end_min}", 1:3: unknown variable accessed: maintenance_day

Which is why I had to shift the calculations into the `.tf` files.